src/profiles_management/pmr/classes.py

```diff
@@ -0,0 +1,133 @@
+"""This package provides classes for validating and defining a PMR.
+
+The goal of these classes is to represent the PMR in a more
+Pythonic way, rather than a dict of vavlues.
+
+The classes should also provide some common helper methods for
+traversing the information of the PMR, but not functions for
+directly affecting the K8s cluster based on the PMR.
+
+This is similar to how the K8s python clients have classes
+for representing the objects and different methods for doing operations
+with them.
+"""
+
+import logging
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, List
+
+import jsonschema
+
+from profiles_management.pmr.schema import RESOURCE_QUOTA_SCHEMA
+
+log = logging.getLogger(__name__)
+
+
+class UserKind(Enum):
+    """Class representing the kind of the user."""
+
+    USER = "user"
+    SERVICE_ACCOUNT = "service-account"
+
+
+class ContributorRole(Enum):
+    """Class representing the role of the user."""
+
+    ADMIN = "admin"
+    EDIT = "edit"
+    VIEW = "view"
+
+
+@dataclass
+class Contributor:
+    """Class representing which users should have access to a Profile."""
+
+    name: str
+    role: ContributorRole
+
+
+@dataclass
+class Owner:
+    """Class representing the owner field of a Profile."""
+
+    name: str
+    kind: UserKind
+
+
+@dataclass
+class Profile:
```

Comment:
Woo Hoo, pydantic ftw!

Will update to use it (am ironing out some details on whether to use `BaseModel` or `dataclass` from pydantic)

Reply:
I updated the code to use Pydantic
630b04f
f170187

The overview of the changes is the following:
1. Used `BaseModel` everywhere, and not `dataclass` (from pydantic) since I wanted to use the `extras` to disallow extra fields (ResourceQuotaSpec) [[1]](https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.extra) [[2]](https://docs.pydantic.dev/latest/concepts/dataclasses/)
2. Added classes for `ResourceQuotaSpec`, so that we do type validation to this field as well

Regarding the `ResourceQuotaSpec`, here's an overview of other options I evaluated:
1. Use `lightkube`'s class. This doesn't work as it's not a Pydantic model and no validation was happening
2. Try to (un-successfully) set a custom [JSON Schema](https://docs.pydantic.dev/latest/concepts/json_schema/) for the validation of the `resources` attribute

Both the above felt hacky and going around pydantic. In the second case especially, which would require us to maintain a json schema and also pydantic couldn't keep this into consideration if in the future it would need to [validate a dict](https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel.model_validate)